vitrina/datasets/views.py

```diff
@@ -85,20 +86,22 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class DatasetDetailView(DetailView):
+class DatasetDetailView(LanguageChoiceMixin, DetailView):
     model = Dataset
     template_name = 'vitrina/datasets/detail.html'
     context_object_name = 'dataset'
 
     def get_context_data(self, **kwargs):
         context_data = super().get_context_data(**kwargs)
         dataset = context_data.get('dataset')
+        dataset.set_current_language(self.request.session.get('LANGUAGE_CODE'))
```

Comment:
I still don't understand why you need this? Please, do not try to override default language settings. Django automatically selects correct language and there is no need to change it manually:

https://docs.djangoproject.com/en/3.2/topics/i18n/translation/#how-django-discovers-language-preference

Parler also uses language set by Django:

> By default, the current Django language is used.
> -- https://django-parler.readthedocs.io/en/stable/quickstart.html#changing-the-language

Anyway, currently, when I try to change language, dataset details page shows english if I select lt, and shows lithuanian if I select en.

This should work correctly without any custom language setting.